addons/product/models/product_template.py

```diff
@@ -410,19 +414,9 @@ def create(self, vals_list):
         # This is needed to set given values to first variant after creation
         for template, vals in zip(templates, vals_list):
             related_vals = {}
-            if vals.get('barcode'):
-                related_vals['barcode'] = vals['barcode']
-            if vals.get('default_code'):
-                related_vals['default_code'] = vals['default_code']
-            if vals.get('standard_price'):
-                related_vals['standard_price'] = vals['standard_price']
-            if vals.get('volume'):
-                related_vals['volume'] = vals['volume']
-            if vals.get('weight'):
-                related_vals['weight'] = vals['weight']
-            # Please do forward port
-            if vals.get('packaging_ids'):
-                related_vals['packaging_ids'] = vals['packaging_ids']
+            for field_name in self._get_related_fields_variant_template() & vals.keys():
+                if vals.get(field_name):
```

Comment:
PLZ not this line.... How ca we reset the default_code or barcode or.... to empty???

Reply:
It might make sense to allow passing falsy values to reset the variant values, but then the PR is not only about adding a hook and keeping the same behavior.

If you always use the template values, then any default (default field attribute, `ir.default`, ...) specified on `product.product` model for those fields won't have any impact unless you create the `product.product` first...

With the `product.template` and `product.product` architecture, a lot of problems like this one have no true solution and compromises are found one way or another ^^.

Reply:
:+1:

Reply:
A bit late to the party, but you could check if `vals.get(field_name)` is not `None`.
`.get` by default returns `None` if it didn't find the value you're looking for, and in case of setting field values to "empty", you could always use `""` or `False`.